telemetry/packages/server/src/modules/measurement/MeasurementReading.types.ts

```diff
@@ -1,10 +1,39 @@
+import { pods } from '@hyped/telemetry-constants';
+import { zodEnumFromObjKeys } from 'src/utils/zodEnumFromObjKeys';
 import { z } from 'zod';
 
-export const MeasurementReadingSchema = z.object({
-  podId: z.string(),
-  measurementKey: z.string(),
-  timestamp: z.string(), // to handle nanoseconds timestamp
-  value: z.number(),
-});
+export const MeasurementReadingSchema = z
+  .object({
+    podId: zodEnumFromObjKeys(pods),
+    measurementKey: z.string(),
+    timestamp: z.string(), // to handle nanoseconds timestamp
+    value: z.number(),
+  })
+  // Validate measurement exists and enum value is valid (if applicable)
+  .refine(
```

Comment:
One thing you could potentially do here as well is validate integers/floats...buttttt that sounds like it might be painful. Could potentially do it 'loose'-ly:

I.e. for integers

`127.1` does not validate
`127.0` does validate
`127` does validate

Reply:
Good idea! Can get the format from the pods object. Will add this :)